web/src/navbar_alerts.ts

```diff
@@ -147,8 +151,31 @@ export function get_demo_organization_deadline_days_remaining(): number {
     return days_remaining;
 }
 
+function is_browser_timezone_same_as(uncanonicalized_target_timezone: string): boolean {
+    if (uncanonicalized_target_timezone === "") {
+        return false;
+    }
+    const canonocalized_browser_timezone = new Intl.DateTimeFormat("en-US", {
+        timeZone: timerender.display_time_zone,
+    }).resolvedOptions().timeZone;
+    const canonicalized_target_timezone = new Intl.DateTimeFormat("en-US", {
+        timeZone: uncanonicalized_target_timezone,
+    }).resolvedOptions().timeZone;
+    return canonocalized_browser_timezone === canonicalized_target_timezone;
```

Comment:
Cool, this logic looks right.

This should get tests. One test case should use two different names for the same time zone, and confirm that they compare as equal — for example Asia/Kolkata and Asia/Calcutta, or Europe/Kyiv and Europe/Kiev.

Another test should confirm we don't have the bug that was discussed [in the chat thread](https://chat.zulip.org/#narrow/stream/378-api-design/topic/Automatically.20detect.20user's.20timezone/near/1921455). To do that, take the three time zones America/Los_Angeles, America/Phoenix, and America/Denver, and confirm that America/Phoenix is different from both of the others. For good measure and to help make the point of the test clear to someone reading the test, confirm that the current time in America/Phoenix *does* equal the current time in one of those two other time zones.

Reply:
I have added some tests.